src/structures/GuildScheduledEvent.js

```diff
@@ -0,0 +1,325 @@
+   * guildScheduledEvent.delete()
+   *  .then(guildScheduledEvent => console.log(guildScheduledEvent))
+   *  .catch(console.error);
+   */
+  async delete() {
+    await this.guild.scheduledEvents.delete(this.id);
+    const clone = this._clone();
+    clone.deleted = true;
+    return clone;
+  }
+
+  /**
+   * Sets a new name for the guild scheduled event.
+   * @param {string} name The new name of the guild scheduled event
+   * @returns {Promise<GuildScheduledEvent>}
+   * @example
+   * // Set name of a guild scheduled event
+   * guildScheduledEvent.setName('Birthday Party')
+   *  .then(guildScheduledEvent => console.log(`Set the name to: ${guildScheduledEvent.name}`))
+   *  .catch(console.error);
+   */
+  setName(name) {
+    return this.edit({ name });
+  }
+
+  /**
+   * Sets a new time to schedule the event at.
+   * @param {DateResolvable} scheduledStartTime The time to schedule the event at
+   * @returns {Promise<GuildScheduledEvent>}
+   * @example
+   * // Set start time of a guild scheduled event
+   * guildScheduledEvent.setScheduledStartTime('September 24, 2022'))
+   *  .then(guildScheduledEvent => console.log(`Set the start time to: ${guildScheduledEvent.scheduledStartTime}`))
+   *  .catch(console.error);
+   */
+  setScheduledStartTime(scheduledStartTime) {
+    return this.edit({ scheduledStartTime });
+  }
+
+  /**
+   * Sets a new time to end the event at.
+   * @param {?DateResolvable} scheduledEndTime The time to end the event at
+   * @returns {Promise<GuildScheduledEvent>}
+   * @example
+   * // Set end time of a guild scheduled event
+   * guildScheduledEvent.setScheduledEndTime('September 25, 2022'))
+   *  .then(guildScheduledEvent => console.log(`Set the end time to: ${guildScheduledEvent.scheduledEndTime}`))
+   *  .catch(console.error);
+   */
+  setScheduledEndTime(scheduledEndTime) {
+    return this.edit({ scheduledEndTime });
+  }
+
+  /**
+   * Sets the new description of the guild scheduled event.
+   * @param {string} description The description of the guild scheduled event
+   * @returns {Promise<GuildScheduledEvent>}
+   * @example
+   * // Set description of a guild scheduled event
+   * guildScheduledEvent.setDescription('A virtual birthday party')
+   *  .then(guildScheduledEvent => console.log(`Set the description to: ${guildScheduledEvent.description}`))
+   *  .catch(console.error);
+   */
+  setDescription(description) {
+    return this.edit({ description });
+  }
+
+  /**
+   * Sets the new status of the guild scheduled event.
+   * @param {GuildScheduledEventStatus|number} status The status of the guild scheduled event
+   * @returns {Promise<GuildScheduledEvent>}
+   * @example
+   * // Set status of a guild scheduled event
+   * guildScheduledEvent.setStatus('ACTIVE')
+   *  .then(guildScheduledEvent => console.log(`Set the status to: ${guildScheduledEvent.status}`))
+   *  .catch(console.error);
+   */
+  setStatus(status) {
+    return this.edit({ status });
```

Comment:
Is there any way we could check for this in the typings at least?
![image](https://user-images.githubusercontent.com/38259440/143689231-68fa0764-394d-4346-b6d5-4d7f382a3273.png)
(taken from the Discord docs, cc @suneettipirneni)

Reply:
If there's a way to know the status of an event compile-time, then it is possible.

Reply:
Well the status property is of type GuildScheduledEventStatus which is one of the following:
![image](https://user-images.githubusercontent.com/38259440/143690164-3d8e3aae-e5d7-4399-8b63-44f913acf365.png)
so I guess it would be possible? Just no clue how

Reply:
I'm saying is there a way to know *which* event status the event has before runtime?

Reply:
Probably with a series of `extends` checks on the status property? Should be something similar to what was done with component collector types right?

Reply:
Moving the convo back here, if you provide a type parameter in the class (`GuildScheduledEvent<Status extends GuildScheduledEventStatus = GuildScheduledEventStatus>`) you can then use that parameter and check if it extends each one of the status types. I think if the type parameter is used as a property it can be inferred, but I'm not sure